Fix styling of ExternalLinks within Cards

matt-west opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Related to DSGCOM-105

## Proposed Changes

* Remove underline override for the ExternalLink component.
* Exempt ExternalLink from link styling within cards so that the default component behaviour is used.
* Remove CSS overrides for correcting underlines on ExternalLinks within Notices.

| Before | After |
|--------|--------|
| <img width="240" alt="Screenshot 2025-05-16 at 10 24 16" src="https://github.com/user-attachments/assets/f4f952c3-ae92-4ef7-98ec-107580f367e4" /> | <img width="240" alt="Screenshot 2025-05-16 at 10 23 43" src="https://github.com/user-attachments/assets/03c46533-99ec-46cc-97bb-13637800766f" /> | 

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* To ensure that the arrow in the ExternalLink component is not underlined.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Checkout this branch.
* Run `yarn start`.
* Navigate to `/me`.
* Observer the gravatar link at the bottom of the form.
* Check for other cases of ExternalLink within the dashboard.

matt-west committed:
Fix styling of ExternalLinks within Cards.

matt-west commented:
@joen @tellthemachines I’d appreciate your eyes on this change given the recent work you did in #101358 to start standardising link behaviour.

Also requesting a review from @tyxla for the change to the wp component overrides.

I’ve gone through the dashboard and can't see any unintended consequences of this change. I’d appreciate your doing a quick test to ensure I didn't miss anything though.

matt-west commented on packages/components/src/card/style.scss:
```diff
@@ -13,10 +13,10 @@
 	border: none;
 
-	 a {
+	a:not(.components-external-link) {
```
I’m not overly happy with this. It feels hacky. Open to cleaner ideas for ensuring the default text-decoration behaviour from the ExternalLink component is maintained.

aduth replied:
Trying to understand the behavior where we want the underline to be hidden, is it strictly associated with the "linked" card, i.e. the `href` prop? In which case, could we avoid dealing with `a` broadly and instead scope it to e.g. `.card.is-card-link a { ... }` or add a custom class for [the anchor](https://github.com/Automattic/wp-calypso/blob/a40795bc0bca860c173f8fd6c0a92bf9c64a1d50/packages/components/src/card/index.tsx#L56)? 

It seems like it would be technically possible to include anchors other than through the `href` prop and without using `<p>` tags that would be styled without underline, but I don't know if we have those, or if we even want that behavior? Looking at [the DevDocs examples](https://wpcalypso.wordpress.com/devdocs/design/card), I don't see any other than the linked cards.

matt-west replied:
There's some additional context on why this was added in the first place in #101358.

We’d like links that appear within text copy to have an underline. I believe this rule was originally added to ensure that only links within text content in a card get underlined. @tellthemachines may be able to provide more background on that though.

aduth replied:
Yeah, what I mean is that if we should want underlines except in rare cases, we can target those rare cases explicitly for the `text-underline` rule, so that we don't have to worry about overriding the rule for external links.

i.e. (diff against `trunk`):

```diff
diff --git a/packages/components/src/card/style.scss b/packages/components/src/card/style.scss
index 304eb2658e1..404eb8474f9 100644
--- a/packages/components/src/card/style.scss
+++ b/packages/components/src/card/style.scss
@@ -13,12 +13,9 @@
 	// Card class name is too generic. Which results and this component randomly getting borders.
 	border: none;
 
-	 a {
+	&.is-card a {
 		text-decoration: none;
 	}
-	p a {
-		text-decoration: underline;
-	}
 
 	@include a8c-clear-fix;
 ```

matt-west replied:
Ah, I’m with you. I think it was implemented this way because there are quite of lot of cases where we don’t want underlines and I presume it was easier to fix that here than update all the individual cases.

For example, if we refactor this as you suggested we end up with erroneous underlines on the feature cards in the site dashboard.

<img width="1175" alt="Screenshot 2025-05-16 at 17 25 23" src="https://github.com/user-attachments/assets/eabc7d74-6a2a-4f5f-aab4-9a24881e841b" />

Links in the site overview:

<img width="1178" alt="Screenshot 2025-05-16 at 17 25 48" src="https://github.com/user-attachments/assets/32824d3c-f743-437e-b2d4-5e32ee6e158a" />

Plugins cards:

<img width="1362" alt="Screenshot 2025-05-16 at 17 27 09" src="https://github.com/user-attachments/assets/ef3759b8-82af-4b40-99a3-5de1b2bedae2" />

That's just a few cases I found with a quick walkthrough. I’m sure there are many others too.

aduth replied:
Thanks for highlighting those examples. I suspected there might be some, though I'm still rebuilding my familiarity with the application 😄  As an aside, my personal opinion is that all of those (except the last) look like the ideal appearance that we should want 😉 🤷 (clarity of links, etc.)

But I think that's a bigger question than we should try to tackle here.

tellthemachines replied:
> Yeah, what I mean is that if we should want underlines except in rare cases, we can target those rare cases explicitly for the text-underline rule, so that we don't have to worry about overriding the rule for external links.

From memory I added `text-decoration: none` to the `is-card-link` class to target those cards that are wrapped in an anchor, so they wouldn't end up with everything inside them underlined. E.g. this card:

<img width="269" alt="Screenshot 2025-05-19 at 2 14 57 pm" src="https://github.com/user-attachments/assets/cdfe3f1f-77bb-4b72-96e0-4d095608e98a" />

The rule was is only inline links should be underlined, and we were starting from a position of having a general override on _all_ links, which I removed, so I was trying to be cautious in not ending up with underlines where we really don't want them 😅 but I do agree having underlines is in most cases better than not having them!

jasmussen reviewed (commented):
I'm close to a 👍 👍 on this one, but I'd like a final look by @tyxla since it's going in the component overrides stylesheet. It seems reasonable enough and a good solution _in absence of using the ExternalLink component_. So curious where this lands.

Thanks, Matt, for the PR!

tyxla commented on client/assets/stylesheets/_wp-components-overrides.scss:
```diff
@@ -57,11 +57,3 @@
 	}
 }
-
-.components-external-link__contents {
-	text-decoration: none;
-
-	&:hover {
-		text-decoration: underline;
-	}
-}
```
Seems like we've overwritten these styles already in some places, which means we'll need to do some cleanup together with this change. Specifically here:

https://github.com/Automattic/wp-calypso/blob/e43d88b62a09adbb9bb7f13cf45bb4e2e76eb5ed/client/me/notification-settings/push-notification-settings/style.scss#L14-L24

matt-west replied:
Thanks. @simison just highlighted this on Linear too. I’ll take a look.

matt-west replied:
Addressed in [d890573](https://github.com/Automattic/wp-calypso/pull/103484/commits/d890573c796d398cef1a40c3eb2a94d2f14905cb).

matt-west committed:
Cleanup external link overrides for notices.

The previous workaround is no longer needed if we remove the explicit text-decoration rules from links within notices.

Links we still be correctly underlined if they appear in the notice text due to upstream rules.

tyxla commented:
Pinging the rest of my team since I'm AFK until Thursday, and I don't want to be blocking this one.

aduth reviewed (commented):
One more instance I wonder if we need to remove `text-underline` styling? 

https://github.com/Automattic/wp-calypso/blob/d890573c796d398cef1a40c3eb2a94d2f14905cb/packages/odie-client/src/components/message/style.scss#L200-L217

I tested this in the branch and I see an underline below the error. Manually disabling the styles gets it to show as expected.

![image](https://github.com/user-attachments/assets/fe936863-1081-40d8-b8f0-d4429a268c63)

To test, I used Cmd+K prompt to "Get help" and then clicked "Still need help?" to start chat conversation. The link above is shown at the bottom of the conversation.

matt-west committed:
Cleanup ExternalLink overrides in support chat.

aduth reviewed (approved):
LGTM
